### PR TITLE
feat: Send messages using Helix API

### LINKED ISSUE #5127: Implement "Send Chat Message" over Helix
https://dev.twitch.tv/docs/api/reference/#send-chat-message

### PR DESCRIPTION
**Pull request by Nerixyz**
This changes the default way messages are sent to use the Helix API instead of IRC. There's still an option to use IRC.

I decided to use a struct as the parameters to the Helix call to make help callers get the order right (using `{.property = value}`), since it would be four `QString` values in a row.

This augments #4962.
Fixes #5127.

### CODE REVIEW

File: src/providers/twitch/api/Helix.cpp [hunk 1]
```diff
@@ -2840,6 +2840,94 @@ void Helix::sendShoutout(
         .execute();
 }
 
+// https://dev.twitch.tv/docs/api/reference/#send-a-shoutout
```
Comment (Wissididom) [inline src/providers/twitch/api/Helix.cpp:2843]:
```suggestion
// https://dev.twitch.tv/docs/api/reference/#send-chat-message
```

Comment (pajlada) [inline src/providers/twitch/api/Helix.cpp:2843]:
```suggestion
// https://dev.twitch.tv/docs/api/reference/#send-chat-message
```

File: src/singletons/Settings.hpp [hunk 2]
```diff
@@ -524,6 +524,8 @@ class Settings
         HelixTimegateOverride::Timegate,
     };
 
+    BoolSetting enableHelixChatSend = {"/misc/helixChatSend", true};
```
Comment (pajlada) [inline src/singletons/Settings.hpp:527]: I would prefer if this is an enum setting with the values:
 - Default (this should be the default value)
 - IRC
 - Helix

**Review by pajlada — COMMENTED** [review]
and as mentioned in chat, the default should be IRC for now

**Review by pajlada — APPROVED** [review]

**Pull request merged**